frontend/src/features/archived-items/archived-item-state-filter.ts

```diff
@@ -69,6 +69,15 @@ export class ArchivedItemStateFilter extends BtrixElement {
         this.selected = new Map();
       }
     }
+    if (changedProperties.has("selectedStates")) {
+      this.dispatchEvent(
+        new CustomEvent<
+          BtrixChangeEvent<ChangeArchivedItemStateEventDetails>["detail"]
+        >("btrix-change", {
+          detail: { value: this.selectedStates },
+        }),
+      );
+    }
   }
```

Comment:
Prefer to have the event dispatched in the updated callback, since `willUpdate` should generally be used to compute and batch updates and `updated` better indicates that the value has been committed.

```suggestion
  }

  protected updated(changedProperties: PropertyValues<this>): void {
    if (changedProperties.has("selectedStates")) {
      this.dispatchEvent(
        new CustomEvent<
          BtrixChangeEvent<ChangeArchivedItemStateEventDetails>["detail"]
        >("btrix-change", {
          detail: { value: this.selectedStates },
        }),
      );
    }
  }
```

Reply:
Sure, makes sense!